v0/destinations/posthog/transform.js

```diff
@@ -0,0 +1,105 @@
+const { CONFIG_CATEGORIES, MAPPING_CONFIG } = require("./config");
+const {
+  defaultRequestConfig,
+  getFieldValueFromMessage,
+  constructPayload,
+  defaultPostRequestConfig
+} = require("../../util");
+const { Message } = require("../../util/enum/message");
+
+// Logic To match destination Property key that is in Rudder Stack Properties Object.
+const generatePropertyDefination = message => {
+  const PHPropertyJson = [
+    { destKey: "$os", sourceKeys: "context.os.name" },
+    { destKey: "browser", sourceKeys: "context.browser.name" },
+    { destKey: "$current_url", sourceKeys: "context.page.url" },
+    { destKey: "$host", sourceKeys: "" },
+    { destKey: "$pathname", sourceKeys: "context.page.path" },
+    { destKey: "$browser_version", sourceKeys: "context.browser.version" },
+    { destKey: "$screen_height", sourceKeys: "context.screen.height" },
+    { destKey: "$screen_width", sourceKeys: "context.screen.width" },
+    { destKey: "$lib", sourceKeys: "context.library.name" },
+    { destKey: "$lib_version", sourceKeys: "context.library.version" },
+    { destKey: "$insert_id", sourceKeys: "" }, // TO DO : Need to Check
+    { destKey: "$time", sourceKeys: ["originalTimestamp", "timestamp"] },
+    { destKey: "$device_id", sourceKeys: "context.device.id" },
+    { destKey: "distinct_id", sourceKeys: "userId" },
+    { destKey: "$initial_referrer", sourceKeys: "properties.referrer" },
+    { destKey: "$initial_referring_domain", sourceKeys: "properties.url" },
+    { destKey: "$ip", sourceKeys: "context.ip" },
+    { destKey: "$timestamp", sourceKeys: ["originalTimestamp", "timestamp"] },
+    { destKey: "$anon_distinct_id", sourceKeys: "anonymousId" }
+  ];
+  // TO DO :: Delete Property that is already defined in above json
+  const data = constructPayload(message, PHPropertyJson);
+
+  return data;
+};
+
+const responseBuilderSimple = (message, category, destination) => {
+  const payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+  if (!payload) {
+    // fail-safety for developer error
+    // eslint-disable-next-line no-undef
+    throw Error(Message.FailedToConstructPayload);
+  }
+
+  if (payload.properties) {
```

Comment:
I don't think we need to handle properties separately unless it's an absolute necessity.